android/java/org/chromium/chrome/browser/crypto_wallet/activities/BuySendSwapActivity.java

```diff
@@ -1381,7 +1381,20 @@ public FilterTextFromToValueText(boolean from) {
         @Override
         public void onTextChanged(CharSequence s, int start, int before, int count) {
             if (mFrom ? mFromValueText.hasFocus() : mToValueText.hasFocus()) {
-                getSendSwapQuota(mFrom, false);
+                boolean inputValid = true;
+                try {
+                    Utils.parseDouble(s.toString());
```

Comment:
you can get rid of `inputValid` variable by moving execution with valid data after `parseDouble`:
```Java
                try {
                    Utils.parseDouble(s.toString());
                    
                    setSendToFromValueValidationResult("", true, !mFrom);
                    getSendSwapQuota(mFrom, false);
                } catch (ParseException e) {
                    Log.w(TAG, e.getMessage(), e);
                    setSendToFromValueValidationResult(
                            getString(R.string.crypto_wallet_error_wrong_number_format), true,
                            !mFrom);
                }
```